### PR TITLE
Silences: Use PostableSilence as the base struct for creating silences

### PR DESCRIPTION
**Pull request by gotjosh**

### CODE REVIEW

File: pkg/api/alertmanager.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ import (
 // swagger:parameters RouteCreateSilence
 type CreateSilenceParams struct {
 	// in:body
```
Comment (gotjosh) [inline pkg/api/alertmanager.go:95]: Upstream uses the same nomenclature but unsure whenever we've deliberately changed it or this was unintended.

```
/ swagger:parameters postSilences
type PostSilencesParams struct {

	// HTTP Request Object
	HTTPRequest *http.Request `json:"-"`

	/*The silence to create
	  Required: true
	  In: body
	*/
	Silence *models.PostableSilence
}
```

File: pkg/api/alertmanager.go [hunk 2]
```diff
@@ -93,26 +93,29 @@ import (
...
+	// in:query
+	Filter []string
 }
 
 // swagger:model
-type GettableSilences amv2.GettableSilences
+type PostableSilence = amv2.PostableSilence
+
+// swagger:model
+type GettableSilences = amv2.GettableSilences
```
Comment (gotjosh) [inline pkg/api/alertmanager.go:115]: From a brief research, I was unsure whenever there would be any downsides of using type aliasing vs struct embedding. In this case, I don't want to add any extra methods so I went with the former. Happy to listed to opinions.

**Review by papagian — CHANGES_REQUESTED** [review]
LGTM, you have to push also the modifications in spec.json and post.json

File: pkg/api/alertmanager.go (hunk 1 above)
Comment (papagian) [reply pkg/api/alertmanager.go:95]: I don't think it was deliberately changed. @domasx2 can you confirm?

**Review by papagian — APPROVED** [review]

**Pull request merged**